source/agora/node/Validator.d

```diff
@@ -555,7 +555,12 @@ public class Validator : FullNode, API
 
         // regenerate the quorums
         if (validators_changed || need_shuffle)
+        {
+            // Try to register immediately if we are a validator for the next cycle
+            if (this.enroll_man.isEnrolled(block.header.height + 1, &this.utxo_set.peekUTXO))
+                () @trusted { this.network.onRegisterName(); } ();
```

Comment:
A `@system` function is a function for which we can't prove that there won't be any memory corruption for any combination of parameter. A `@safe` function is one where we can prove it. A `@trusted` lambda is usually used to signify that the call is `@safe` with this set of parameters.

This is easier to demonstrate with C function, let's say `strdup`.
If you do:
```D
void main () @safe
{
    auto ptr = () @trusted { return strdup("Hello World"); }();
}
```

A human can easily look at this, and knowing that string literals are always `\0`-terminated, the call will not result in any memory corruption.

Now when I look at this block of code, I am confused, because `onRegisterName` does not take any parameter.
So the only way that it could be `@system` and proven `@safe` here would be if calling `onRegisterName` was not `@safe` depending on the state of `this`. That would obviously be a very big breach of encapsulation.

So it seems to me that in this case, `onRegisterName` should have been made `@safe`, instead of being wrapped here.